File: src/main/java/dev/jason/project/spring/vc_server/data/dto/RegisterUserDto.java
package dev.jason.project.spring.vc_server.data.dto;

import dev.jason.project.spring.vc_server.domain.model.User;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

public record RegisterUserDto(String uid, String displayName, String profilePictureUrl, DeviceDto device) {

    public User toDomainUser() {
        return new User(
            uid,
            displayName,
            profilePictureUrl,
            device == null ? new ArrayList<>(List.of()) : new ArrayList<>(List.of(device.toDomainModel(LocalDateTime.now()))),
            new ArrayList<>(List.of()),
            new ArrayList<>(List.of()),
            User.Status.Online,
            System.currentTimeMillis()
        );
    }
}

File: messaging/src/main/java/dev/jason/app/compose/vaultchat/messaging/data/dto/RegisterUserDto.kt
package dev.jason.app.compose.vaultchat.messaging.data.dto

import dev.jason.app.compose.vaultchat.messaging.domain.model.RegisterUser
import kotlinx.serialization.Serializable

@Serializable
data class RegisterUserDto(
    val uid: String,
    val displayName: String,
    val profilePictureUrl: String,
    val fcmToken: String
)

fun RegisterUser.toDto() = RegisterUserDto(uid, displayName, profilePictureUrl, token)